components/style/properties/helpers/animated_properties.mako.rs

```diff
@@ -155,6 +155,25 @@ pub trait Interpolate: Sized {
     fn interpolate(&self, other: &Self, time: f64) -> Result<Self, ()>;
 }
 
+/// https://drafts.csswg.org/css-transitions/#animtype-repeatable-list
+pub trait RepeatableListInterpolate: Interpolate {}
+
+impl<T: RepeatableListInterpolate> Interpolate for Vec<T> {
+    fn interpolate(&self, other: &Self, time: f64) -> Result<Self, ()> {
+        use num_integer::lcm;
+        let len = lcm(self.len(), other.len());
+        let ret = self.iter().cycle().zip(other.iter().cycle())
+                      .take(len)
+                      .filter_map(|(ref me, ref you)| {
```

Comment:
Note that `Result<Vec<T>, E>` implements `FromIterator<Result<T, E>>`, so this is better written as:

```rust
        self.iter().cycle().zip(other.iter().cycle()).take(len).map(|(me, you)| {
            me.interpolate(you, time)
        }).collect()